### DISCUSSION TITLE
Error: “mutation metadata length does not match derived state length”

### DISCUSSION THREAD

**Discussion by Fganau** [tskit-dev/pyslim#265] [Q&A]
Hi

Hi,
I have recently moved from SLIM 3.6 to 3.7 (as I couldn’t make it to work on 3.6 either). I want to recapitate a SLIM tree sequence adding neutral mutations with msprime. I am studying selective sweeps and I want to use this recapitated population as a replacement for the burn-in period required to reach mutation-drift equilibrium before selective sweeps are implemented onto the population.

When I load the .trees file into SLIM, though, I get the error: “mutation metadata length does not match derived state length”. Any ideas on how to solve it? 
The code I use is:
```
import msprime
import tskit
import numpy as np
import pyslim


CL=100 #Chromosome length
RecRate=0.5
MutRate=1e-5
Ne_pop = 1000
seed_1 = 30
seed_2 = 31
breaks = [0, CL/3, CL*3/5, CL*9/10,CL] #Breaking points for different chromosome sgements in the subsequent selective sweep analysis with SLIM


recomb_map = msprime.RateMap(
  position = breaks,
  rate = [RecRate,RecRate,RecRate,RecRate])



ts=pyslim.load("TreeSequence_path”)
recap_ts = pyslim.recapitate(ts, 
                             recombination_rate = recomb_map,
                            ancestral_Ne=Ne_pop, random_seed=seed_1)

Out=pyslim.SlimTreeSequence(msprime.mutate(recap_ts, rate=MutRate)) recap_sampled_mutated_ts
Out.dump("Output_treeSeq_path")
```

Thanks! I started using pyslim and tskit a few weeks ago, so I am not exactly agile working with it…

**Comment by petrelharp** [comment]
So - you want to load the tree sequence produced at the end of the script above into SLiM? If so, then you need to use the SLiMMutationModel in msprime - see [the pyslim docs](https://tskit.dev/pyslim/docs/latest/tutorial.html#adding-neutral-mutations-to-a-slim-simulation) for an example of how to do that. Let me know if this works?
